pkg/boot/kexec/elf_rel_load.go

```diff
@@ -0,0 +1,39 @@
+package kexec
+
+import (
+	"bytes"
+	"debug/elf"
+	"fmt"
+)
+
+// RelocateAndLoad does ...
+func RelocateAndLoad(purgatory []byte, purgatorySize uint, min uint, max uint, end int, flags uint32) (entry uintptr, err error) {
+	pgtory, err := elf.NewFile(bytes.NewReader(purgatory))
+	if err != nil {
+		return uintptr(0), fmt.Errorf("parse elf file from purgatory buffer: %v", err)
+	}
+
+	if pgtory.Version != elf.EV_CURRENT { // Should always be 1.
```

Comment:
can you have these print out the values that are acceptable? It's a real slog, when given a message like this, to figure out exactly what it wanted. Further, it would be nice to provide a summary of ALL the things that did not match, instead of just the FIRST thing that did not match. This can be done with something like
var err error
if blah blah {

see error wrapping examples in
https://play.golang.org/p/NL_MiNC31o6

Reply:
Done.

I realized I had to remove them, as `debug/elf` already does all of those checks before returning a `elf.File`. So if there is any issue with the kernel code, what `debug/elf` returned will be presented to user. https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/debug/elf/file.go;l=240 looks reasonable.